File: lib/data/mock/mock_profile_repository.dart
import '../../core/roles.dart';
import '../models/user_profile.dart';
import '../repositories/profile_repository.dart';
import '_latency.dart';
import 'mock_data.dart';

class MockProfileRepository implements ProfileRepository {
  bool simulateError = false;

  @override
  Future<UserProfile> getProfile(UserRole activeRole) async {
    await simulateLatency();
    if (simulateError) {
      throw Exception('Mock: failed to load profile');
    }
    return MockData.profileForRole(activeRole);
  }
}

File: Data/Mock/MockProfileRepository.cs
using OrquestadorFrontend.Core;
using OrquestadorFrontend.Data.Models;
using OrquestadorFrontend.Data.Repositories;

namespace OrquestadorFrontend.Data.Mock;

public sealed class MockProfileRepository : IProfileRepository
{
    public bool SimulateError { get; set; }

    public async Task<UserProfile> GetProfileAsync(UserRole activeRole)
    {
        await MockLatency.SimulateAsync();
        if (SimulateError) throw new InvalidOperationException("Mock: failed to load profile");
        return MockData.ProfileForRole(activeRole);
    }
}
